src/debugger/android/androidPlatform.ts

```diff
@@ -19,13 +20,56 @@ export class AndroidPlatform implements IAppPlatform {
 
     public enableJSDebuggingMode(runOptions: IRunOptions): Q.Promise<void> {
         let pkg = new Package(runOptions.projectRoot);
-
         return pkg.name()
             .then(appName => new PackageNameResolver(appName).resolvePackageName(runOptions.projectRoot))
             .then(packageName => {
-                let enableDebugCommand = `adb shell am broadcast -a "${packageName.toLowerCase()}.RELOAD_APP_ACTION" --ez jsproxy true`;
+                let deviceResolver = new DeviceResolver();
                 let cexec = new CommandExecutor(runOptions.projectRoot);
-                return cexec.execute(enableDebugCommand);
+                let debugTarget: string = null;
+                return deviceResolver.getConnectedDevices()
+                    .then((devices: IDevice[]) => {
+                        if (devices.length > 1) {
+                            /* more than one device or emulator */
+                            debugTarget = this.getTargetEmulator(runOptions, devices);
+                        }
+                    })
+                    .then(() => {
+                        if (debugTarget) {
+                            /* Launching is needed only if we have more than one device available */
+                            let launchAppCommand = `adb -s ${debugTarget} shell am start -n com.awesomeproject/.MainActivity`;
```

Comment:
I see that line as:
 'shell', 'am', 'start', '-n', packageName + '/.MainActivity'